### DISCUSSION TITLE
How can I change the prompt depending on the shell being used?

### DISCUSSION THREAD

**Discussion by eggbean** [starship/starship#5126] [Q&A] [answered]
I currently have this in my config for bash prompts:

```
[character]
success_symbol = "[\\$](white)"
error_symbol = "[\\$](bold red)"
```

I want to use a `%` instead of a `$` for when I am using zsh instead of bash, as a reminder of which shell I am using. I use the same dotfiles on all hosts, and sometimes zsh will not be available, so this would be a nice way of reminding me, but how do I do it?

**Comment by eggbean** [comment]
Okay, I just realised that I can set `$STARSHIP_CONFIG` to a different file for zsh. However, that means I will have two files to update, which would be the same apart from this character. I suppose that's not difficult to manage, but is there any other way?

I'm thinking of something like setting `$STARSHIP_CONFIG` to point to different files for just this setting, and those files to 'include' a third shared .toml file for the rest of the configuration. Is this possible? Is it possible to set includes or source directives in .toml files?

**Comment by mickimnet** [comment, marked as answer]
If you always only have **one** shell running at the same time, you could use `starship config character.success_symbol "[\\$](white)"` in your Bash config and `starship config character.success_symbol "[\\%](white)"` in your Zsh config. Be aware, that `starship config` changes the values in the `starship.toml` file.

If you use Bash and Zsh at the same time, you need to use two configuration files and use something like the following in your shell configuration:

```bash
...
## change Starship configuration for Zsh
cp ~/.config/starship.toml ~/.config/starship-zsh.toml
export STARSHIP_CONFIG="$HOME/.config/starship-zsh.toml"
starship config character.success_symbol "[\\%](white)"
starship config character.error_symbol "[\\%](bold red)"
eval "$(starship init zsh)"
...
```

I hope that helps.

EDITH: fixed configuration snippet.

**Comment by eggbean** [comment]
Thanks. Neat solution. I'll just point out that the `%` characters don't need to be escaped, as like that it errors.

```
## change Starship configuration for Zsh
cp ~/.config/starship.toml ~/.config/starship-zsh.toml
export STARSHIP_CONFIG="$HOME/.config/starship-zsh.toml"
starship config character.success_symbol "[%](white)"
starship config character.error_symbol "[%](bold red)"
eval "$(starship init zsh)"
```
